### ISSUE TITLE
salt cloud no bootstrap after Azure provisioning

### ISSUE THREAD

**Issue by cskowronnek** [saltstack/salt#48966]
### Description of Issue/Question
Bootstrap fails after provisioning a Linux VM on Azure using

 `salt-cloud -p azure Node-X`

This feature worked fine before I updated Salt to the latest development release.

It seems that Salt can't login after creating the node. I tried both public/private key and password authentication. I can login via SSH and key or password from the master after creation. 

Log entry debug:
```
[DEBUG   ] Deploying 10.xx.xx.xx at 2018-08-07 11:10:59
[DEBUG   ] Attempting connection to host 10.xx.xx.xx on port 22
[DEBUG   ] SSH port 22 on 10.xx.xx.xx is available
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 1 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 2 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 3 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 4 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 5 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 6 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 7 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 8 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 9 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 10 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 11 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 12 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 13 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 14 of 15)
[DEBUG   ] Using /etc/salt/cloud.profiles.d/cs.key as the key_filename
[DEBUG   ] Attempting to authenticate as skowronc (try 15 of 15)
[ERROR   ] Failed to start Salt on host Node-X
```

    ----------
    Error:
        ----------
        Not Deployed:
            Failed to start Salt on host Node-X
```
/var/log/salt/cloud:

`2018-08-07 10:45:07,869 [salt.loaded.int.utils.cloud:658 ][ERROR   ][44087] Failed to start Salt on host Node-X`

### Setup
/etc/salt/cloud with:

```
script: bootstrap-salt
start_action: state.apply

```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2018.3.0

Dependency Versions:
           cffi: 1.11.2
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-131-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by gtmanfred** [comment]
Can you login to the minion and check the syslog and see if there are any error messages about why the user trying to login is being rejected?

Also, try setting the log level to `trace` `salt.utils.cloud.bootstrap` should be logging the commands it is running, and it would be helpful to see those.

**Comment by cskowronnek** [comment]
Thanks for your reply. Log level trace brought nothing new.
The auth.log of the minion showed no login tries. The only corresponding line I found was:

`Did not receive identification string from <ip of salt master>`

**Comment by gtmanfred** [comment]
did you run salt-cloud with -l trace? because the ssh commands used to try and connect should be provided there, it sounds like salt is not passing in an ssh key.

**Comment by cskowronnek** [comment]
Yes, I did. -l trace added only a few lines to the log, nothing to do with this step.

I tried it with ssh_key and ssh_pass, neither worked. Manual login after VM creation works fine, so I don't see a firewall or any other problem. As there are no login attempts it seems the SSH command was never called.

**Comment by gtmanfred** [comment]
WIthout seeing the ssh commands, i cannot know why it is failing to login.

Please sanatize that log, and then share it, so that I can take a look.

Thanks,
Daniel

**Comment by cskowronnek** [comment]
That's 13k lines, with a lot of information to sanitize. I grep'ped the log and there are no ssh connection string. Below is all what log level TRACE added. Maybe I'm switching back to the latest release. I don't know what kind of regression happened in the latestet develop branch.

```
[TRACE   ] The required configuration section, 'fluent_handler', was not found the in the configuration. Not loading the fluent logging handlers module.
[TRACE   ] None of the required configuration sections, 'logstash_udp_handler' and 'logstash_zmq_handler', were found in the configuration. Not loading the Logstash logging handlers module.
[TRACE   ] Trying generic group list for 'root'
[TRACE   ] Group list for user 'root': []
[TRACE   ] 'parallels.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] Error loading clouds.libvirt: Unable to locate or import python libvirt library.
[TRACE   ] loaded json json lib
[TRACE   ] Error loading clouds.virtualbox: The virtualbox driver cannot be loaded: 'vboxapi' is not installed.
[TRACE   ] 'parallels.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] 'proxmox.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] 'parallels.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] Error loading clouds.libvirt: Unable to locate or import python libvirt library.
[TRACE   ] Error loading clouds.virtualbox: The virtualbox driver cannot be loaded: 'vboxapi' is not installed.
[TRACE   ] 'parallels.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] 'proxmox.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] IPCClient: Connecting to socket: /var/run/salt/master/master_event_pull.ipc
[TRACE   ] Attempting to obtain lock for /var/cache/salt/cloud/index.p
[TRACE   ] Removing lock for /var/cache/salt/cloud/index.p
[TRACE   ] IPCClient: Connecting to socket: /var/run/salt/master/master_event_pull.ipc
[TRACE   ] IPCClient: Connecting to socket: /var/run/salt/master/master_event_pull.ipc
[TRACE   ] IPCClient: Connecting to socket: /var/run/salt/master/master_event_pull.ipc
```

**Comment by gtmanfred** [comment]
Here is the log message your above debug command shows.

https://github.com/saltstack/salt/blob/2018.3/salt/utils/cloud.py#L940

It should then try to run `date`  using `root_cmd` right below it, which will setup some logging commands.

https://github.com/saltstack/salt/blob/2018.3/salt/utils/cloud.py#L2145

Then it constructs a whole big ssh command, and adds it to the logging command

https://github.com/saltstack/salt/blob/2018.3/salt/utils/cloud.py#L2238

Then it logs it

https://github.com/saltstack/salt/blob/2018.3/salt/utils/cloud.py#L2246

And finally runs it

https://github.com/saltstack/salt/blob/2018.3/salt/utils/cloud.py#L2248

So there should be a more than just 

```
[DEBUG   ] Attempting to authenticate as skowronc (try x of 15)
```
From salt.utils.cloud:bootstrap if this is from develop.  That is the information I need to troubleshoot this.

Can you try clearing out .pyc files?

**Comment by cskowronnek** [comment]
I've no clue why, but it works again now.
I tried clearing the pyc-files. No result, I tried to trace the lines you mentioned, but found no reason why nothing is logged.
As a shot in the dark I downloaded the latest develop code and tried that. Now it's working again. I don't see any changes between the two checkouts that could've caused the problem.

Thanks for your effort, I think this can be closed.
